src/test/java/co/rsk/federate/signing/hsm/advanceblockchain/HsmBookkeepingClientImplTest.java

```diff
@@ -168,151 +155,55 @@ public void getChunks_keepPreviousChunkLastItem_false() {
         assertEquals(result.get(1)[2] + 1, result.get(2)[0].intValue());
     }
 
-    @Test
-    public void sendBlockHeadersChunks_no_data() throws HSMClientException {
-        hsmBookkeepingClient.sendBlockHeadersChunks(null, "", false);
-        hsmBookkeepingClient.sendBlockHeadersChunks(Collections.emptyList(), "", false);
-        hsmBookkeepingClient.sendBlockHeadersChunks(null, null, false);
-        hsmBookkeepingClient.sendBlockHeadersChunks(Collections.emptyList(), null, false);
-
-        verifyNoInteractions(jsonRpcClientMock);
-    }
-
-    @Test
-    public void sendBlockHeadersChunks_advanceBlockchainInProgress() throws HSMClientException, JsonRpcException {
-        when(jsonRpcClientMock.send(buildVersionRequest())).thenReturn(buildResponse(0, VERSION_TWO));
-        when(jsonRpcClientMock.send(buildExpectedRequest("blockchainState", VERSION_TWO)))
-            .thenReturn(buildResponse(true));
-
-        hsmBookkeepingClient.sendBlockHeadersChunks(Collections.singletonList(""), null, false);
-
-        // once to get the version, second one to verify the blockchain state
-        verify(jsonRpcClientMock, times(2)).send(any(JsonNode.class));
-    }
-
     @Test(expected = HSMBlockchainBookkeepingRelatedException.class)
-    public void sendBlockHeadersChunks_chunk_fails() throws HSMClientException, JsonRpcException {
-        when(jsonRpcClientMock.send(buildVersionRequest())).thenReturn(buildResponse(0, VERSION_TWO));
-        when(jsonRpcClientMock.send(buildExpectedRequest("blockchainState", VERSION_TWO)))
-            .thenReturn(buildResponse(false));
-        when(jsonRpcClientMock.send(any(JsonNode.class))).thenReturn(buildResponse(-203));
-
-        hsmBookkeepingClient.setMaxChunkSizeToHsm(2);
-        hsmBookkeepingClient.sendBlockHeadersChunks(Collections.singletonList(""), null, false);
+    public void updateAncestorBlock_when_HSM_is_stopped() throws HSMClientException {
```

Comment:
According to sonar this test case is not covered. Maybe it's not reaching that condition and throwing an exception earlier?

Reply:
Updated